tests/onebox.go

```diff
@@ -221,11 +247,17 @@ func newTemporal(t *testing.T, params *TemporalParams) *temporalImpl {
 		spanExporters:                    params.SpanExporters,
 		tlsConfigProvider:                params.TLSConfigProvider,
 		captureMetricsHandler:            params.CaptureMetricsHandler,
-		dcClient:                         testDCClient,
+		dcClient:                         dynamicconfig.NewMemoryDCClient(dynamicconfig.NewNoopClient()),
 		serviceFxOptions:                 params.ServiceFxOptions,
 		taskCategoryRegistry:             params.TaskCategoryRegistry,
 	}
-	impl.overrideHistoryDynamicConfig(t, testDCClient)
+	for k, v := range staticOverrides {
+		impl.overrideDynamicConfigByKey(t, k, v)
```

Comment:
I'm not sure if this is better or worse, but just to consider:
The `fallback` for the memory client can be `dynamicconfig.StaticClient(staticOverrides)`, then you only need to set the ones from params and history ones.